token/transfer-hook/cli/src/main.rs

```diff
@@ -0,0 +1,405 @@
+
+    let mut ixs = vec![];
+    if transfer_lamports > 0 {
+        ixs.push(system_instruction::transfer(
+            &payer.pubkey(),
+            &extra_account_metas_address,
+            transfer_lamports,
+        ));
+    }
+    ixs.push(initialize_extra_account_meta_list(
+        program_id,
+        &extra_account_metas_address,
+        token,
+        &mint_authority.pubkey(),
+        &extra_account_metas,
+    ));
+
+    let mut transaction = Transaction::new_with_payer(&ixs, Some(&payer.pubkey()));
+    let blockhash = rpc_client
+        .get_latest_blockhash()
+        .await
+        .map_err(|err| format!("error: unable to get latest blockhash: {err}"))?;
+    let mut signers = vec![payer];
+    if payer.pubkey() != mint_authority.pubkey() {
+        signers.push(mint_authority);
+    }
+    transaction
+        .try_sign(&signers, blockhash)
+        .map_err(|err| format!("error: failed to sign transaction: {err}"))?;
+
+    rpc_client
+        .send_and_confirm_transaction_with_spinner(&transaction)
+        .await
+        .map_err(|err| format!("error: send transaction: {err}").into())
+}
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let app_matches = Command::new(crate_name!())
+        .about(crate_description!())
+        .version(crate_version!())
+        .subcommand_required(true)
+        .arg_required_else_help(true)
+        .arg({
+            let arg = Arg::new("config_file")
+                .short('C')
+                .long("config")
+                .value_name("PATH")
+                .takes_value(true)
+                .global(true)
+                .help("Configuration file to use");
+            if let Some(ref config_file) = *solana_cli_config::CONFIG_FILE {
+                arg.default_value(config_file)
+            } else {
+                arg
+            }
+        })
+        .arg(
+            Arg::new("fee_payer")
+                .long("fee-payer")
+                .value_name("KEYPAIR")
+                .validator(|s| is_valid_signer(s))
+                .takes_value(true)
+                .global(true)
+                .help("Filepath or URL to a keypair to pay transaction fee [default: client keypair]"),
+        )
+        .arg(
+            Arg::new("verbose")
+                .long("verbose")
+                .short('v')
+                .takes_value(false)
+                .global(true)
+                .help("Show additional information"),
+        )
+        .arg(
+            Arg::new("json_rpc_url")
+                .short('u')
+                .long("url")
+                .value_name("URL")
+                .takes_value(true)
+                .global(true)
+                .value_parser(parse_url_or_moniker)
+                .help("JSON RPC URL for the cluster [default: value from configuration file]"),
+        )
+        .subcommand(
+            Command::new("create-extra-account-metas")
```

Comment:
That's a big ol' command (`spl-transfer-hook create-extra-account-metas ...`)!

Reply:
And to think, I shortened it from `create-extra-account-metas-list`. I couldn't come up with anything better, but if you have a suggestion, please let me know!

Reply:
I think it's fine! Best I could come up with is `transfer-hook init-extra-metas`.

Reply:
How about we split the difference with `create-extra-metas`? `init` typically means that you're *just* writing to an allocated + assigned account, and here we're also doing the allocation + assign

Reply:
Leaving that totally up to you, but that sounds good to me!